### PR TITLE
defect #1089493: Combobox now closes if scrolling outside of it

### PR DESCRIPTION
**Pull request by pikachugb**

### CODE REVIEW

File: octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/EntityFieldsComposite.java [hunk 1]
```diff
@@ -134,6 +137,9 @@ public void expansionStateChanged(ExpansionEvent expansionEvent) {
         PrefereceChangeHandler prefereceChangeHandler = () -> drawEntityFields(entityModel);
         PluginPreferenceStorage.addPrefenceChangeHandler(PreferenceConstants.SHOWN_ENTITY_FIELDS, prefereceChangeHandler);
         addDisposeListener(e -> PluginPreferenceStorage.removePrefenceChangeHandler(PreferenceConstants.SHOWN_ENTITY_FIELDS, prefereceChangeHandler));
+        
+        //initialize fieldEditors
+        fieldEditors = new ArrayList<FieldEditor>();
```
Comment (vladbenchea) [inline octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/EntityFieldsComposite.java:142]: Is there a reason why you didn't do this when declaring the variable? I see that you're not waiting for any other value to be computed before initializing this param. It's a better code practice to try to initialize where you're declaring, or closer to where you're declaring the variable.

File: octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/util/EntityComboBox.java [hunk 2]
```diff
@@ -93,6 +93,8 @@ public void mouseExit(MouseEvent mouseEvent) {
     private boolean isLoadingIndicatorEnabled = true;
     private boolean isFilteringEnabled = true;
     
+    public static ArrayList<EntityComboBox> entityComboBoxInstances = new ArrayList<EntityComboBox>();
```
Comment (vladbenchea) [inline octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/util/EntityComboBox.java:96]: Change from ArrayList to List, which is an interface, while ArrayList is an implementation. 
https://stackoverflow.com/questions/2279030/type-list-vs-type-arraylist-in-java
Check here the first two answers for a better explanation why this is preffered.

Comment (pikachugb) [reply octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/util/EntityComboBox.java:96]: Forgot to delete that line of code :D

File: octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/EntityFieldsComposite.java [hunk 3]
```diff
@@ -134,6 +137,9 @@ public void expansionStateChanged(ExpansionEvent expansionEvent) {
         PrefereceChangeHandler prefereceChangeHandler = () -> drawEntityFields(entityModel);
         PluginPreferenceStorage.addPrefenceChangeHandler(PreferenceConstants.SHOWN_ENTITY_FIELDS, prefereceChangeHandler);
         addDisposeListener(e -> PluginPreferenceStorage.removePrefenceChangeHandler(PreferenceConstants.SHOWN_ENTITY_FIELDS, prefereceChangeHandler));
+        
+        //initialize fieldEditors
```
Comment (alexandra-marinescu-microfocus) [inline octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/EntityFieldsComposite.java:141]: remove comment :) the code is too simple to document what it does

File: octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/field/FieldEditor.java [hunk 4]
```diff
@@ -18,4 +18,6 @@
 
 public interface FieldEditor extends Drawable {
     public void setField(EntityModelWrapper entityModel, String fieldName);
+    
+    public default void closeEntityComboBox() {};
```
Comment (alexandra-marinescu-microfocus) [inline octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/field/FieldEditor.java:22]: this class does not look to me like the best place to add this method

File: octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/field/EntityComboBoxFieldEditor.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+package com.hpe.octane.ideplugins.eclipse.ui.entitydetail.field;
+
+import org.eclipse.swt.graphics.Drawable;
+
+public interface EntityComboBoxFieldEditor extends Drawable {    
+    public default void closeEntityComboBox() {};
```
Comment (alexandra-marinescu-microfocus) [inline octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/field/EntityComboBoxFieldEditor.java:18]: no need for default

**Review by alexandra-marinescu-microfocus — APPROVED** [review]

**Pull request merged**